### ISSUE TITLE
The subscription isn't deleted after the client disconnected with clean session = true @ v0.8

### ISSUE THREAD

**Issue by esmusssein** [moquette-io/moquette#143]
Suppose a client connected with clean session = true, subscribed topic `foo` then disconnected. Then if someone publishes to topic `foo`, the broker throws NullPointerException because the subscription of the previous client is still alive but its session was cleaned.

I wrote a simple test of it at `ServerIntegrationEmbeddedPublishTest.java`

```
@Test
public void testClientSubscribeAfterDisconnected() throws Exception {
    MQTT mqtt = new MQTT();
    mqtt.setHost("localhost", 1883);
    mqtt.setClientId("subscriber-1");
    BlockingConnection subscriber = mqtt.blockingConnection();
    subscriber.connect();
    subscriber.subscribe(new Topic[]{new Topic("foo", QoS.AT_MOST_ONCE)});
    subscriber.disconnect();
    internalPublishToWithQosAndRetained("foo", QOSType.MOST_ONE, false);
    verifyMessageIsReceivedSuccessfully();
}
```

**Comment by LiangCan** [comment]
Appears Yes, I edited "moquette" source and played breakpoint debugging of the connection through disconnected client can still be found in the queue, but the implementation of a message list cleanup client object sent a
 "java.lang .ClassCastException "

**Comment by andsel** [comment]
Hi @esmusssein you are embedding the broker inside another application? Because when you write "Then if someone publishes to topic" this let me think it's a publish from an external client, but using internalPublishToWithQosAndRetained you are publishing internally from the broker.

 Andrea

**Comment by esmusssein** [comment]
Hello @andsel, I meant `someone` as the internal publisher as well as an external client.

**Comment by andsel** [comment]
Hi @esmusssein I've foudn the bug, was due to a publish from the broker to a not more connected client. Your test  should be like

``` java
    @Test(expected = EOFException.class)
    public void testClientSubscribeAfterDisconnected() throws Exception {
        LOG.info("*** testClientSubscribeAfterDisconnected ***");

        subscribeToWithQos("foo", QoS.AT_MOST_ONCE);
        m_subscriber.disconnect();

        internalPublishToWithQosAndRetained("foo", QOSType.MOST_ONE, false);

        //verifyMessageIsReceivedSuccessfully();
        m_subscriber.receive(2, TimeUnit.MILLISECONDS);
    }
```

The subscription client is with clean session, and you was trying to publish on it after the subscribing client disconnected.
Fixed with dfa75e145fe8a7fc9266826ab5ba0e3288992a6a
